index.js

```diff
@@ -1 +1,93 @@
 // implement your API here
+const express = require('express');
+
+const db = require('./data/db.js');
+
+const server = express();
+
+server.use(express.json());
+
+server.get('/', (req, res) => {
+    res.send("Server setup in progress...");
+});
+
+server.post('/api/users', (req, res) => {
+    const newUser = req.body;
+    if (!name || !bio) {
+        res.status(400).json({ errorMessage: "Please provide name and bio for the user." })
+    }
+    db
+        .insert(newUser)
+        .then(user => {
+            res.status(201).json(user);
+        })
+        .catch(err => {
+            res.status(500).json({ error: "There was an error while saving the user to the database" })
+        });
+});
+
+server.get('/api/users', (req, res) => {
+    db
+        .find()
+        .then(users => {
+            res.status(200).json({users});
+        })
+        .catch(err => {
+            res.status(500).json({ error: "The users information could not be retrieved." })
+        });
+});
+
+server.get('/api/users/:id', (req, res) => {
+    userId = req.params.id
+    db
+     .findById(userId)
+     .then(user => {
+         if (user.length === 0) {
+             res.status(404).json({ error: "User with that id is not found"});
+             return;
+         }
+         res.status(200).json(user);
+     })
+     .catch(err => {
+         res.status(500).json({ error: "The user information could not be retrieved." })
+     });
+});
+
+server.delete('/api/users/:id', (req, res) => {
+    const userId = req.params.id
+    db
+        .remove(userId)
+        .then(userId => {
+            if (userId.length === 0) {
```

Comment:
using userId.length won't work here since userId stands for the result of whatever db.remove(userId) returned. it will either return 1 or 0 depending if it could delete or not. no matter what, that's a length of 1, so you'll never hit this 404 even if they try to use a user id that doesn't exist. you could say `if (userId === 0)` since that only happens if it failed